assets/src/edit-story/components/header/title.js

```diff
@@ -25,26 +27,42 @@ const Input = styled.input`
 
 function Title() {
 	const {
-		state: { story: { title, status } },
+		state: { story: { title, slug } },
 		actions: { updateStory },
 	} = useStory();
 
+	const { storyId } = useConfig();
+
 	const handleChange = useCallback(
 		( evt ) => updateStory( { properties: { title: evt.target.value } } ),
 		[ updateStory ],
 	);
 
+	const handleBlur = useCallback(
+		() => {
+			if ( ! slug || slug === storyId ) {
+				const clearSlug = encodeURIComponent( cleanForSlug( titleFormatted( title ) ) ) || storyId;
+				updateStory( { properties: { slug: clearSlug } } );
+			}
+		}, [ slug, storyId, title, titleFormatted, updateStory ],
+	);
+
+	// TODO Make sure that Auto Draft checks translations.
+	const titleFormatted = useCallback(
+		( rawTitle ) => {
+			return ( rawTitle === 'Auto Draft' ) ? '' : rawTitle;
+		}, [],
+	);
+
 	if ( typeof title !== 'string' ) {
 		return null;
 	}
 
-	// TODO Make sure that Auto Draft checks translations.
-	const titleFormatted = ( status === 'auto-draft' && title === 'Auto Draft' ) ? '' : title;
-
 	return (
 		<Input
-			value={ titleFormatted }
+			value={ titleFormatted( title ) }
```

Comment:
Doesn't this give warnings about switching from controlled to uncontrolled or is that not an issue?

Reply:
Can you explain this a little. I haven't seen any issue as of yet.

Reply:
Looks like title is never `null`? So there's no controlled/uncontrolled issue?

Reply:
Seems to work fine, my bad.